index.mjs

```diff
@@ -179,68 +192,70 @@ export default class MediaMixin extends Componentry.Module {
       }
     })
 
-    router.put('/media/stage/:system?',async (req, res) => {
```

Comment:
This changes the external facing API of the module. See bluefire/components/AdCreatives.mjs. This component is the GUI for media management. It is a fork of media-mixin/components/InputMultipleImages.mjs. There are a few other clients dependent on media-mixin, though none I couldn't change if we must.

Stage uses the database to store the pending properties of a new upload. In the past we then continued to use the database to mirror the object in S3. This turned out to be unnecessary as we can just query S3 itself. However, I think we still need to stage the upload, in memory perphaps (though we have to consider multiple servers providing this code behind a load balancer. It's fine to use the db for temporary shared storage, or redis or something). By staging the upload we can then track progress events fired by XMLHttpRequest (fetch doesn't fire upload events). See media-mixin/components/InputMultipleImages.mjs line 256. This is important for large files that otherwise freeze the UI until upload is complete. Further, staging gets a onetime use url for uploading straight to S3.

If there is a better way to upload large files with a progress indicator, I'm open to suggestions, but we shouldn't be changing the API on which clients depend unless we must. Regardless, the goal of this project is for the GUI presented by components/AdCreatives.mjs to allow the user to upload images, rotate images and select/modify crop presets.

Reply:
So I see several similarities between AdCreatives.mjs and InputMultipleImages.mjs. So we need stage for uploading files and to display progress of uploading, don't we ? Now in my local bluefire, the uploading works with out stage and progress of uploading works. So why we need db for uploading ?. I'll find out the ways to display progress.

bluefire/components/AdCreatives.mjs
<img width="727" alt="Screenshot 2024-10-28 at 13 30 02" src="https://github.com/user-attachments/assets/15c691ed-0f89-4ca4-a921-07625df1f871">

Reply:
Please check my fixes. I use db only for stage. Stage payload will be kept in db temporary, then after uploading this data will be deleted.

Reply:
Also there were some changes in metric-im/componentry/index.mjs

<img width="1500" alt="Screenshot 2024-10-30 at 12 37 58" src="https://github.com/user-attachments/assets/a94de17b-2018-4961-8c34-93b64e25ea2f">


bluefire/index.mjs
<img width="1500" alt="Screenshot 2024-10-30 at 12 38 48" src="https://github.com/user-attachments/assets/ffeb47aa-4cc2-4046-92f6-7c050d7099ad">